java/kotlin-extractor2/src/main/kotlin/entities/FunctionalInterface.kt

```diff
@@ -0,0 +1,569 @@
+            builtinTypes.int,
+            pIdx,
+            locId,
+            arrayAccessId,
+            1,
+            funLabels.methodId,
+            enclosingStmtId
+        )
+    }
+}
+
+
+/**
+ * Gets the labels for functions belonging to
+ * - local functions, and
+ * - lambdas.
+ */
+private fun KotlinFileExtractor.getLocallyVisibleFunctionLabels(f: KaAnonymousFunctionSymbol): LocallyVisibleFunctionLabels {
+    if (!f.isLocal) {
+        logger.error("Extracting a non-local function as a local one")
+    }
+
+    return tw.lm.getOrAddLocallyVisibleFunctionLabelMapping(f) {
+        val classId = tw.getFreshIdLabel<DbClassorinterface>()
+        val javaResult = TypeResult(classId/* , "TODO", "TODO" */)
+        val kotlinTypeId =
+            tw.getLabelFor<DbKt_class_type>("@\"kt_class;{$classId}\"") {
+                tw.writeKt_class_types(it, classId)
+            }
+        val kotlinResult = TypeResult(kotlinTypeId /* , "TODO", "TODO" */)
+
+        LocallyVisibleFunctionLabels(
+            type = TypeResults(javaResult, kotlinResult),
+            constructor = tw.getFreshIdLabel(),
+            constructorBlock = tw.getFreshIdLabel(),
+            function = tw.getFreshIdLabel()
+        )
+    }
+}
+
+/**
+ * Extracts the class around a local function or a lambda. The superclass must have a no-arg
+ * constructor.
+ */
+context(KaSession)
+private fun KotlinFileExtractor.extractGeneratedClass(
+    localFunction: KaFunctionSymbol,
+    elementToReportOn: PsiElement,
+    superTypes: List<KaType>,
+    compilerGeneratedKindOverride: CompilerGeneratedKinds? = null // OLD: KE1
+): Label<out DbClassorinterface> {
+    val ids = tw.lm.getLocallyVisibleFunctionLabelMapping(localFunction)
+
+    val id =
+        extractGeneratedClass(
+            ids,
+            superTypes,
+            tw.getLocation(elementToReportOn),
+            elementToReportOn,
+            compilerGeneratedKindOverride = compilerGeneratedKindOverride
+            /*
+            OLD: KE1
+            localFunction.parent,
+             */
+        )
+
+    // Extract local function as a member
+    extractFunction(
+        localFunction,
+        id,
+        /*
+        OLD: KE1
+        extractBody = true,
+        extractMethodAndParameterTypeAccesses = true,
+        extractAnnotations = false,
+        null,
+        listOf()
+         */
+    )
+
+    return id
+}
+
+/** Extracts the class around a local function, a lambda, or a function reference. */
+context(KaSession)
+@OptIn(KaExperimentalApi::class)
+private fun KotlinFileExtractor.extractGeneratedClass(
+    ids: GeneratedClassLabels,
+    superTypes: List<KaType>,
+    locId: Label<DbLocation>,
+    elementToReportOn: PsiElement,
+    compilerGeneratedKindOverride: CompilerGeneratedKinds? = null,
+    superConstructorSelector: (KaFunctionSymbol) -> Boolean = { it.valueParameters.isEmpty() },
+    extractSuperConstructorArgs: (Label<DbSuperconstructorinvocationstmt>) -> Unit = {},
```

Comment:
Can we just inline the default here too?

Reply:
I think this can now be removed

Reply:
Ah, sorry, I missed that. Anyway, like you say, we can always add it back if we need it.